528.py

```diff
@@ -0,0 +1,42 @@
+# Runtime: O(log n)
+# n = len(w)
+#
+# Space: O(n)
+# n = len(w)
+class Solution:
+    def __init__(self, w: List[int]):
+        self.total = 0
+        self.ranges = []
```

Comment:
ok I see that you called this `.ranges` which is totally fine. This approach is optimal. 

This is a common example of something called `prefix_sum` . The prefix sum array is the sum of all the previous elements in `w`

```
nums = [1,2,3]
prefix_sum = [1,3,6] or [0,1,3,6]
```
Depending on how you want to index. Just note that this is a commonly used term that can be applied to many problems. 

https://leetcode.com/problems/subarray-sum-equals-k/description/